p2p/src/msg.rs

```diff
@@ -612,3 +616,42 @@ impl Readable for TxHashSetArchive {
 		})
 	}
 }
+
+pub struct KernelDataRequest {}
+
+impl Writeable for KernelDataRequest {
+	fn write<W: Writer>(&self, _writer: &mut W) -> Result<(), ser::Error> {
+		Ok(())
+	}
+}
+
+pub struct KernelDataResponse {
+	/// Support for multiple versions of data serialization.
+	/// 0 - the original "fixed size" kernel serialization format (lock_height and fee required)
+	/// 1 - "variable size" kernel serialization
+	///    * lock_height only included in HeightLocked kernels
+	///    * fee omitted for coinbase kernels
+	///
+	/// Note: Both versions will produce identical hash values (and corresponding MMR root).
+	/// Only the "data" serialization differs.
+	///
+	pub version: u8,
```

Comment:
Not sure this will be very useful. If the peer that requested the kernel data can't support v1, then it shouldn't ever receive v1 kernel data. It should be up to the requester to ask which version it needs.

Reply:
Also, if we implement the protocol versioning scheme I suggested here: https://github.com/mimblewimble/grin-pm/issues/102#issuecomment-482751069 , we won't even need a version byte at all.